logpyle/__init__.py

```diff
@@ -1096,11 +1111,18 @@ class DependencyData(Record):
 
         return parsed, dep_data
 
+    def _calculate_next_watch_step(self) -> None:
+        ticks_per_sec = (self.tick_count/max(1, time()-self.start_time)
+                            * self.watch_interval)
```

Comment:
This still isn't right IMO. You need to compute the rate since the last change in watch interval.

Reply:
I've played around with computing the rate when the interval changes, but that leads to a super-jerky watch printing at the interval changes (especially when the interval is reduced such that a watch is printed very soon after the change) that it is imho not worth it.

Reply:
Could you push the example you came up with?

Reply:
Just curious.. can an interface where the user just calls `start_tick`, `end_tick` get wedged into this?  This would be real easy for the user to control vs. even setting a fixed interval.

Reply:
> Could you push the example you came up with?

https://github.com/illinois-ceesd/logpyle/pull/66/commits/14300874118456271c4cabe554e4693263d36796 is one example I had.

Reply:
> Just curious.. can an interface where the user just calls `start_tick`, `end_tick` get wedged into this? This would be real easy for the user to control vs. even setting a fixed interval.

Hmm, Im not sure what you mean - could you please clarify?

Reply:
> > Just curious.. can an interface where the user just calls `start_tick`, `end_tick` get wedged into this? This would be real easy for the user to control vs. even setting a fixed interval.
> 
> Hmm, Im not sure what you mean - could you please clarify?

Such that when the user calls `end_tick`, the watches print with elapsed time since `start_tick`?

Reply:
Yikes, you were right the whole time. Sorry. ticks/s can be estimated independent of watches. I've reverted it to that in 
acb0c2e. If you like that, it's good to go from my end. Sorry again about all the waste of time here.

Reply:
I just tweaked the watch intervals a bit so that CI is a bit faster, but it looks good to go for me.